### PR TITLE
Logging module

### PR DESCRIPTION
**Pull request by ryanrolds**
Adds logging module discussed in [#1390](https://github.com/adaptlearning/adapt_framework/issues/1399).

### CODE REVIEW

File: src/core/js/logging.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+define(['core/js/adapt'], function(Adapt) {
```
Comment (oliverfoster) [inline src/core/js/logging.js]: could you put some whitespace into this file?

should be spaces 4 indentation, line breaks between function definitions etc

File: src/core/js/logging.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+      // Query string log level wins over the config
+      if (this._override) {
+        this._config._level = this._override;
+      }
+
+      this.debug('Logging config loaded');
+      this.trigger('log:ready');
+    },
+    debug: function(msg, data) {
```
Comment (oliverfoster) [inline src/core/js/logging.js]: these functions need their parameters to behaviour more like the ``console.log`` function.
essentially each argument forms part of the output string.

```javascript
    fatal: function() {
      this._log('fatal', [].array.slice.call(arguments, 0)));
    },
    _log: function(level, datas) {
      if (!this._config._isEnabled) {
        return;
      }

      if (this._levels[level] < this._levels[this._config._level]) {
        return;
      }

      if (this._config._console) {
        var log = [level.toUpperCase() + ':'];
        data && log.push.apply(log, datas);

        console.log.apply(console, log);
      }

      // Allow error reporting plugins to hook and report to logging systems
      this.trigger('log:' + level, datas);
    }
```

Comment (ryanrolds) [inline src/core/js/logging.js]: Will do.

File: src/core/js/logging.js (hunk 1 above)
Comment (ryanrolds) [inline src/core/js/logging.js]: The 4 for indentation conflicts with: https://github.com/adaptlearning/documentation/blob/master/01_cross_workstream/style_guide.md which is linked in https://github.com/adaptlearning/adapt_framework/wiki/Contributing-code

Comment (oliverfoster) [inline src/core/js/logging.js]: Ignore the referenced/copied standard, it hasn't been formalized yet. It should say "spaces 4" indentation, no tabs, etc.

**Comment by ryanrolds** [comment]
* Fixed the indentation
* Adjusted log function arguments
* Query string override is now case-insensitive
* Added some blank lines
* Comments

Should be ready to review again.

**Comment by ryanrolds** [comment]
IE8 has been EOL for a while now and as far as I'm aware is not supported on current MS OSs.

**Comment by ryanrolds** [comment]
For ethical reasons I don't make changes to specifically support IE8, the browser is EOL and no longer receiving security updates, it's usage is negligent.

**Comment by oliverfoster** [comment]
Fair enough. I'll see if someone else can take over.

**Comment by oliverfoster** [comment]
functionality added on #1425 

thanks for doing this @ryanrolds 
nice bit of code

**Pull request closed without merging**